src/dmd/expressionsem.d

```diff
@@ -8093,6 +8093,16 @@ private extern (C++) final class ExpressionSemanticVisitor : Visitor
             exp = new ConstructExp(exp.loc, exp.e1, exp.e2);
             exp.type = t;
 
+            // @@@DEPRECATED_2020-06@@@
+            // When removing, alter `checkModifiable` to return the correct value
+            // the correct value.
+            if (sc.func.isStaticCtorDeclaration() && !sc.func.isSharedStaticCtorDeclaration() &&
```

Comment:
Can both these condition ever be true?

Reply:
Yes, because the later inherits from the former. They are also function declarations.